### ISSUE TITLE
Investigate Native bootstrapping errors in macos

### ISSUE THREAD

**Issue by jakubstilec** [dotnet/arcade#7327]
Contact: @joeloff 

Description: Two Darwin builds are consistently failing within seconds of being kicked off - [logs](https://dev.azure.com/dnceng/public/_build/results?buildId=1117229&view=logs&jobId=912d8f18-7352-5eae-79f7-5bf7a7d4e42c&j=912d8f18-7352-5eae-79f7-5bf7a7d4e42c&t=7bc63554-b1c8-5333-abd3-b4356d3b4aed)

```
Downloading https://netcorenativeassets.blob.core.windows.net/resource-packages/external/macos/cmake/cmake-3.16.4-MacOS-x86_64.tar.gz 
extracting from /Users/runner/.netcoreeng/native/temp/cmake-3.16.4-MacOS-x86_64.tar.gz to /Users/runner/work/1/s/.tools/bin/cmake/3.16.4 
Extracting /Users/runner/.netcoreeng/native/temp/cmake-3.16.4-MacOS-x86_64.tar.gz to /Users/runner/work/1/s/.tools/bin/cmake/3.16.4 
Creating directory: '/Users/runner/work/1/s/.tools/bin/cmake/3.16.4' 
Extracting archive 
Generating '/Users/runner/work/1/s/.tools/bin/cmake.sh' shim 
/Users/runner/work/1/s/eng/common/native/common-library.sh: line 151: Write-PipelineTelemetryError: command not found 
/Users/runner/work/1/s/eng/common/native/install-cmake.sh: line 113: Write-PipelineTelemetryError: command not found 
Execution Failed 
##[error]Bash exited with code '1'. 
```

The function is defined in eng\common\pipeline-logging-functions.sh

Affected code is under https://github.com/dotnet/arcade/tree/main/eng/common/native

**Comment by riarenas** [comment]
I'm not sure any repos have ever attempted to actually use the native bootstrapping on a mac. I'll take a look to see if there's something obvious that we can fix, or if we just need better bootstrapping support in general (Which would be part of this epic: https://github.com/dotnet/arcade/issues/4072 and probably out of scope for a quick FR fix)

**Comment by joeloff** [comment]
What does the runtime repo do? They have to build for MacOS right? And they use cmake. I'm not aware of a way to exclude the package from global.json.

The project that's blocked is needed for .NET 6 setup. What would the alternative be? Creating a new repo where we build the executable for windows and then flow that into the installer repo?

**Comment by riarenas** [comment]
The runtime repo currently does not use the native bootstrapping scripts.

One of the main reasons they are unable to use it is due to currently not having a way to specify the native dependencies per OS: https://github.com/dotnet/arcade/issues/2673

#4072 came to existence precisely because the current simplistic mechanism is not quite good enough for complex scenarios.

**Comment by riarenas** [comment]
```
/Users/runner/work/_temp/8d45f890-2440-4bbd-aa70-7786a958670e.sh: line 1: _AdditionalBuildParameters: command not found
/Users/runner/work/1/s/eng/common/init-tools-native.sh: line 13: declare: -A: invalid option
```
Makes me think the wrong bash is being used here, and the fact that this error is happening but not failing the script might be the reason why the imports of other scripts are not working well.

**Comment by joeloff** [comment]
Turning it off though did unblock my PR

**Comment by riarenas** [comment]
>Turning it off though did unblock my PR

Thanks for the update!

**Comment by jonfortescue** [comment]
Using native tools bootstrapping on macs is an unsupported scenario to my knowledge, and native tools bootstrapping is going away anyway I'm going to be closing this.
